syntax = "proto3";

package media;

import "google/protobuf/empty.proto";

service Media {
    rpc UploadAvatar (UploadAvatarRequest) returns (UploadAvatarResponse);
    rpc GetFileUrl (GetFileUrlRequest) returns (GetFileUrlResponse);
    rpc DeleteFile (DeleteFileRequest) returns (DeleteFileResponse);
    rpc HealthCheck (google.protobuf.Empty) returns (HealthCheckResponse);
}

message UploadAvatarRequest {
    string owner_id = 1;
    bytes file = 2;
    string content_type = 3;
}

message UploadAvatarResponse {
    string id = 1;
    string url = 2;
}

message GetFileUrlRequest {
    string id = 1;
}

message GetFileUrlResponse {
    string url = 1;
}

message DeleteFileRequest {
    string file_id = 1;
    string user_id = 2;
    bool is_superuser = 3;
}

message DeleteFileResponse {
    string status = 1;
}

message HealthCheckResponse {
    string status = 1;
    map<string, string> checks = 2;
}